GoodreadsEDA/bk_plt.py

```diff
@@ -1,98 +1,131 @@
-        mv_release_datetime = datetime.strptime(mv_release, "%d %B %Y")
-        plt.text(mv_release_datetime + timedelta(days=10), 4,
-                                                 'Movie Release Date',
-                                                 rotation=90,
-                                                 color='#767a78')
-
-        plt.xlabel("Year")
-        plt.xticks(rotation=45)
-        plt.xlim(mv_release_datetime - relativedelta(years=1),
-                 mv_release_datetime + relativedelta(years=1))
-        plt.title(f"Review Scores of '{bk}' over time.")
-        plt.subplots_adjust(bottom=0.2)
-        sns.despine()
-        plt.show()
-    except ValueError as e:
-        print(e)
-        print(df)
-        continue
+
+def plot_results(df, n_reviews, book_title, mv_release):
+    """Plots the relationship between movie release and book ratings over time.
+
+    Args:
+        df (pandas.DataFrame): dataframe containing the values to plot.
+        n_reviews (int): denominator to use for calculating rolling average.
+        book_title (str): title of the book to plot.
+        mv_release (str): the release date of the movie of the book.
+    """
+
+    # getting movie release date
+    mv_release_datetime = datetime.strptime(mv_release, "%d %B %Y")
+    # setting plotting parameters
+    set_plotting_params()
+
+    # plotting rolling average of book reviews over time and date of movie
+    # release
+    df.plot(x=df.index, y='review_score_rolling', color="#f99f75")
+    plt.axvline(x=mv_release, color="#f99f75")
+
+    # adjusting plot labels and orientation
+    plt.title(f"Review Scores of '{book_title}' over time.")
+    plt.yticks(np.arange(2.5, 5.5, .5))  # books are seldom given a score < 3.
+    plt.ylabel(f"Average Rating Per {n_reviews} Reviews")
+    plt.text(mv_release_datetime + timedelta(days=10),
+             4,
+             'Movie Release Date',
+             rotation=90,
+             color='#767a78')
+    plt.xlabel("Time")
+    plt.xticks(rotation=45)
+    plt.xlim(mv_release_datetime - relativedelta(years=1),
+             mv_release_datetime + relativedelta(years=1))
+    plt.subplots_adjust(bottom=0.2)
+
+    sns.despine()
+    plt.show()
+
+
+def main(movie_titles, n_reviews_rolling_avg):
+    """Iterates through movies/book pairs, plotting each one's popularity as a
+    book relative to its release date as a movie.
+
+    Args:
+        movie_titles ([str]): a sequence of strings of the titles of movies that
+           were also books.
+        n_reviews_rolling_avg (int): number of ratings to use when
+          calculating the rolling average rating.
+
+    Raises:
+        ValueError: I suppose this would be to account for an error in reading the SQL query into a DataFrame, not
+          entirely sure.
+    """
+
+    for title in movie_titles:
+        try:
+            # Get df of reviews for the book
+            with sqlite3.connect(f"{os.getcwd()}/review_dbs/reviews.db") as conn:
```

Comment:
This is a good way to make sure your DB connections are closed - `with` exits whatever object was created from the initial statement when the indented block ends. It is a shorter and safer way of managing connections to objects - files, DBs, whatev.